cmd/cstor-pool-mgmt/controller/new-pool-controller/handler.go

```diff
@@ -0,0 +1,245 @@
+	apis2 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha2"
+
+	pool "github.com/openebs/maya/pkg/cstor/pool/v1alpha2"
+	corev1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+)
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the cStorPoolUpdated resource
+// with the current status of the resource.
+func (c *CStorPoolController) syncHandler(key string, operation common.QueueOperation) error {
+	csp, err := c.getCSPObjFromKey(key)
+	if err != nil {
+		return err
+	}
+
+	return c.eventHandler(operation, csp)
+}
+
+// cStorPoolEventHandler is to handle cstor pool related events.
+func (c *CStorPoolController) eventHandler(operation common.QueueOperation, csp *apis2.CStorNPool) error {
+	switch operation {
+	case common.QOpAdd:
+		return c.addEventHandler(csp)
+	case common.QOpDestroy:
+		return c.destroyEventHandler(csp)
+	case common.QOpSync:
+		return c.syncEventHandler(csp)
+	case common.QOpModify:
+		return c.modifyEventHandler(csp)
+	}
+
+	return errors.Errorf("Invalid event {%s} for cStorPool {%s}", operation, csp.Name)
+}
+
+func (c *CStorPoolController) addEventHandler(csp *apis2.CStorNPool) error {
+	var status string
+	var err error
+	var isImported bool
+
+	uid := string(csp.GetUID())
+	importedPool := pool.ImportedCStorPools[uid]
+
+	// TODO
+	// In which scenario it is possible??
+	if importedPool != nil {
+		c.recorder.Event(importedPool,
+			corev1.EventTypeWarning,
+			string(common.AlreadyPresent),
+			string(common.MessageResourceAlreadyPresent))
+		glog.Warningf("Object is already present", string(csp.ObjectMeta.Name))
+	}
+
+	// take a lock for common
+	// TODO move return to label and unlock
+	common.SyncResources.Mux.Lock()
+
+	status, isImported = pool.IsPoolImported(csp, true /* wait if Zrepl is initializing */)
+	if isImported {
+		c.recorder.Event(csp,
+			corev1.EventTypeNormal,
+			string(common.AlreadyPresent),
+			string(common.MessageResourceAlreadyPresent))
+		goto updatestatus
+	}
+
+	// try to import pool
+	status, isImported, err = pool.Import(csp)
+	if isImported {
+		if err != nil {
+			c.recorder.Event(csp,
+				corev1.EventTypeWarning,
+				string(common.FailureImported),
+				string(common.FailureImportOperations))
+			glog.Errorf("Failed to handle add event: import succeeded with failed operations %v", err.Error())
+		} else {
+			c.recorder.Event(csp,
+				corev1.EventTypeNormal,
+				string(common.SuccessImported),
+				string(common.MessageResourceImported))
+		}
+		goto updatestatus
+	}
+
+	//TODO: Why pool validation? should we consider zpool status during sync op only?
+
+	// IsInitStatus is to check if initial status of cstorpool object is `init`.
+	if pool.IsEmptyStatus(csp) || pool.IsPendingStatus(csp) {
+		err = pool.Create(csp)
```

Comment:
As per idiomatic maya practices, I guess this would have been `csp.ZFSClient().Create(cspObj)` where `csp` is the import alias for `github.com/openebs/maya/pkg/cstor/pool/v1alpha2`.